isites_migration/utils.py

```diff
@@ -0,0 +1,348 @@
+            'zip_file_importer',
+            settings_file_url=export_file_url,
+            settings_folder_id=root_folder['id']
+        ).json()
+        progress_url = response['progress_url']
+        logger.info(
+            "Created Canvas content migration %s for file import from %s to Canvas course %s",
+            progress_url,
+            export_file_url,
+            canvas_course_id
+        )
+    except Exception:
+        message = "Failed to complete file import for keyword %s", keyword
+        logger.exception(message)
+        raise RuntimeError(message)
+
+    return progress_url
+
+
+def lock_canvas_folder(canvas_course_id, folder_name):
+    import_folder = _get_import_folder(canvas_course_id, folder_name)
+    try:
+        files.update_folder(
+            SDK_CONTEXT,
+            import_folder['id'],
+            import_folder['name'],
+            import_folder['parent_folder_id'],
+            None,
+            None,
+            'true',
+            None,
+            None
+        )
+        logger.info("Locked import folder %s for canvas_course_id %s", folder_name, canvas_course_id)
+    except CanvasAPIError:
+        logger.exception("Failed to lock import folder %s for canvas_course_id %s", folder_name, canvas_course_id)
+        raise
+
+
+def get_previous_isites_keywords(course_instance_id):
+    """
+    Given a course_instance_id, finds iSite keywords mapped to previous offerings of the course
+
+    Adapted from:
+    http://subversion.icommons.harvard.edu/filedetails.php?
+    repname=Perl.CourseManager&path=%2Fbranches%2Fmod_perl2_2%2Flib%2FCourseTools%2FNewCourseSiteWizard.pm
+
+    :param course_instance_id:
+    :return: The set of iSite keywords
+    """
+    previous_keywords = set()
+    try:
+        course_instance = CourseInstance.objects.get(course_instance_id=course_instance_id)
+    except CourseInstance.DoesNotExist:
+        return previous_keywords
+
+    # Collect the iSites keywords associated with previous offerings of the given course instance
+    course = course_instance.course
+    previous_instance_ids = {c.course_instance_id for c in course.course_instances.all()}
+    # If this is an EXT course, also look for instances where the registrar_code
+    # is stored in the registrar_code_display field
+    if course.school.school_id == 'ext':
+        previous_courses = Course.objects.filter(school=course.school, registrar_code_display=course.registrar_code)
+        previous_instance_ids = previous_instance_ids | {
+            c.course_instance_id for c in CourseInstance.objects.filter(course__in=previous_courses)
+        }
+
+    evm_sql_query = """
+    SELECT course_instance_id, user_id
+    FROM enrollment_viewer_manager
+    WHERE
+    course_instance_id = %s
+    """
+    cursor = connections['termtool'].cursor()
+
+    # Get a list of the enrollment viewer managers for the given course instance
+    cursor.execute(evm_sql_query % course_instance_id)
```

Comment:
I think this needs be done like so:
```
cursor.execute(evm_sql_query, [course_instance_id])
```
to obtain the [param protection offered by Django](https://docs.djangoproject.com/en/1.8/topics/db/sql/#passing-parameters-into-raw).